Framework/MDAlgorithms/src/LoadMD.cpp

```diff
@@ -345,11 +360,25 @@ void LoadMD::loadHisto() {
                       << '\n';
 
   if (pref_QConvention != m_QConvention) {
+    std::vector<double> scaling(m_numDims);
+    for (size_t d = 0; d < m_numDims; d++) {
+      std::string dimd = ws->getDimension(d)->getName();
+      std::string dim2 = dimd.substr(0, 2);
+      dimd = dimd.substr(0, 1);
+
+      // Assume the Q dimensions are those that have names starting with [
+      // such as [H,0.5H,0], or Q_ such as Q_sample_x.
+      // The change in sign should apply only to those.
+      if (dim2 == "Q_" || dimd == "[")
+        scaling[d] = -1.0;
+      else
+        scaling[d] = 1.0;
+    }
```

Comment:
Instead of repeating this code, can we move it into a separate function that is called twice?